subcmd/edit.go

```diff
@@ -0,0 +1,218 @@
+}
+
+type editCmd struct {
+	helped bool
+	editor string
+}
+
+func (cmd *editCmd) ProhibitRootExecution(args []string) bool { return true }
+
+func (cmd *editCmd) FlagSet() *flag.FlagSet {
+	fs := flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	fs.SetOutput(os.Stdout)
+	fs.Usage = func() {
+		fmt.Print(`
+Usage
+  volt edit [-help] [-e|--editor {editor}] {repository} [{repository2} ...]
+
+Quick example
+  $ volt edit tyru/caw.vim # will open the plugconf file for tyru/caw.vim for editing
+
+Description
+  Open the plugconf file(s) of one or more {repository} for editing.
+
+  If the -e option was given, use the given editor for editing those files (unless it cannot be found)
+
+  It also calls "volt build" afterwards if modifications were made to the plugconf file(s).` + "\n\n")
+		//fmt.Println("Options")
+		//fs.PrintDefaults()
+		fmt.Println()
+		cmd.helped = true
+	}
+	fs.StringVar(&cmd.editor, "editor", "", "Use the given editor for editing the plugconf files")
+	fs.StringVar(&cmd.editor, "e", "", "Use the given editor for editing the plugconf files")
+	return fs
+}
+
+func (cmd *editCmd) Run(args []string) *Error {
+	reposPathList, err := cmd.parseArgs(args)
+	if err == ErrShowedHelp {
+		return nil
+	}
+	if err != nil {
+		return &Error{Code: 10, Msg: "Failed to parse args: " + err.Error()}
+	}
+
+	hasChanges, err := cmd.doEdit(reposPathList)
+	if err != nil {
+		//FIXME: Which error code to use?
+		return &Error{Code: 15, Msg: "Failed to edit plugconf file: " + err.Error()}
+	}
+
+	// Build opt dir
+	if hasChanges {
+		err = builder.Build(false)
+		if err != nil {
+			return &Error{Code: 12, Msg: "Could not build " + pathutil.VimVoltDir() + ": " + err.Error()}
+		}
+	}
+
+	return nil
+}
+
+func (cmd *editCmd) doEdit(reposPathList []pathutil.ReposPath) (bool, error) {
+	// Read lock.json
+	lockJSON, err := lockjson.Read()
+	if err != nil {
+		return false, err
+	}
+
+	// Read config.toml
+	cfg, err := config.Read()
+	if err != nil {
+		return false, errors.New("could not read config.toml: " + err.Error())
+	}
+
+	viitor, err := cmd.identifyEditor(cfg)
+	if err != nil || viitor == "" {
+		//FIXME: Which error code to use?
+		return false, &Error{Code: 30, Msg: "No usable viitor found"}
+	}
+
+	changeWasMade := false
+	//FIXME: Run single vim instance? Leads to problems if the configured
+	//editor does not support to open multiple files
+	for _, reposPath := range reposPathList {
+
+		// Edit plugconf file
+		plugconfPath := reposPath.Plugconf()
+
+		// Install a new template if none exists
+		if !pathutil.Exists(plugconfPath) {
+			getCmd := new(getCmd)
+			logger.Debugf("Installing new plugconf for '%s'.", reposPath)
+			getCmd.downloadPlugconf(reposPath)
+		}
+
+		// Remember modification time before opening the editor
+		info, err := os.Stat(plugconfPath)
+		if err != nil {
+			return false, err
+		}
+		mTimeBefore := info.ModTime()
+
+		// Call the editor with the plugconf file
+		vimCmd := exec.Command(viitor, plugconfPath)
```

Comment:
Please rename to `editorCmd`.
But I agree no one should use an editor other than vim when using vim package manager 😂